src/components/AddItem.js

```diff
@@ -24,9 +27,27 @@ const AddItem = ({ token }) => {
         });
   };
 
-  const handleSubmit = (event) => {
+  const handleSubmit = async (event) => {
     event.preventDefault();
-    db.collection(token).add(formData);
+    const itemNameInput = formData.itemName;
+
+    // check to see if the itemName already exists in the user's list - convert input to lowercase and remove any punctuation & white space
+    const cleanInput = itemNameInput.toLowerCase().replace(/[\W_]*|/g, '');
+
+    // query the collection and filter for match - if match exists send the user an error message "this item already exists in your list..." & don't allow them to submit it
+    const queryCollection = db.collection(token);
+    const snapshot = await queryCollection
+      .where('cleanedUpItemName', '==', cleanInput)
+      .get();
+    if (!snapshot.empty) {
+      setHidden(true);
+      return;
+    }
```

Comment:
This totally works...nice!  

Another way to approach it would be to use the results that you already available to you (without making a new call to the db...would need to be passed down via context or props), and then either `filter` or `find` if there is a duplicate item. You would need to do the same `.toLowerCase().replace(/[\W_]*|/g, '')` on the `itemName` of the result item to compare. 

If you all end up re-factoring to pass down state via context or props, this would be possible.